app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -51,31 +52,43 @@ open class TwoPaneFragment : Fragment() {
     fun isOnlyLeftShown() = slidingPaneLayout.let { it.isSlideable && !it.isOpen }
     fun isOnlyRightShown() = slidingPaneLayout.let { it.isSlideable && it.isOpen }
 
+    private val searchFolder by lazy {
+        Folder().apply {
+            id = FolderController.SEARCH_FOLDER_ID
+            name = getString(R.string.searchFolderName)
+        }
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setupSlidingPane()
-        observeFolderName()
+        observeFolderChange()
         observeThreadEvents()
     }
 
     private fun setupSlidingPane() {
         slidingPaneLayout.lockMode = SlidingPaneLayout.LOCK_MODE_LOCKED
     }
 
-    private fun observeFolderName() = with(mainViewModel) {
+    private fun observeFolderChange() = with(mainViewModel) {
 
         Utils.waitInitMediator(currentFolder, isInSearch).observe(viewLifecycleOwner) { (folder, isInSearch) ->
 
-            val displayedFolder = (if (isInSearch) {
-                Folder().apply { name = getString(R.string.searchFolderName) }
+            val currentFolder = if (isInSearch) {
+                searchFolder
             } else {
-                folder
-            }) ?: return@observe
+                folder ?: return@observe
+            }
```

Comment:
That logic can be in the `ViewModel`. That way, you only need to return the folder.